ecosystem/sep-0010.md

```diff
@@ -138,18 +141,21 @@ To validate the challenge transaction the following steps are performed by the s
 * decode the received input as a base64-urlencoded XDR representation of Stellar transaction envelope;
 * verify that transaction source account is equal to the server's signing key;
 * verify that transaction has time bounds set, and that current time is between the minimum and maximum bounds;
-* verify that transaction contains a single [Manage Data](https://www.stellar.org/developers/guides/concepts/list-of-operations.html#manage-data) operation that:
-  * has a non-null source account
 * verify that transaction envelope has a correct signature by server's signing key;
-* if the operation's source account exists:
+* verify that transaction operations:
+  * are all [Manage Data](https://www.stellar.org/developers/guides/concepts/list-of-operations.html#manage-data) operations
+  * have non-null source accounts
+* if the first operation's source account exists:
   * verify that client signatures count is one or more;
-  * verify that client signatures on the transaction are signers of the operation's source account;
+  * verify that client signatures on the transaction are signers of the first operation's source account;
   * verify that client signatures are correct;
   * verify that client signatures provide weight that meets the required threshold(s), if any;
-* if the operation's source account does not exist:
+* if the first operation's source account does not exist:
   * verify that client signature count is one;
   * verify that client signature is correct for the master key of the account address;
 * verify that transaction containing additional Manage Data operations have their source account as the server signing key;
+* if the transaction contains a Manage Data operation with a key containing `attribution`:
+  * verify that the client domain included in the key has a `stellar.toml` file with a `SIGNING_KEY` used to sign the challenge. An internal mapping from client domain to SIGNING_KEY may be used.
```

Comment:
I think we need to go into more detail here, can we model this off the similar change in SEP-10 v3.1 and be explicit about what each field of the operation needs to contain, especially the source account field?

Reply:
I adjusted the Manage Data op definition, let me know if I need to make more adjustments